src/Functions/FunctionHelpers.h

```diff
@@ -108,8 +108,10 @@ struct FunctionArgumentDescriptor
 {
     const char * argument_name;
 
-    std::function<bool (const IDataType &)> type_validator_func;
-    std::function<bool (const IColumn &)> column_validator_func;
+    using TypeValidator = bool (*)(const IDataType &);
+    TypeValidator type_validator_func;
+    using ColumnValidator = bool (*)(const IColumn &);
+    ColumnValidator column_validator_func;
```

Comment:
The old interface used to support any kind of callable (not just c++ functions), was it excessive?